lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformColumnExtensions.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+
+namespace MwbExporter\Formatter\Doctrine2\Annotation\OnCorps;
+
+
+use MwbExporter\Formatter\Doctrine2\Annotation\Model\Column;
+use MwbExporter\Formatter\Doctrine2\Annotation\Model\Table;
+use MwbExporter\Writer\WriterInterface;
+
+trait ApiPlatformColumnExtensions
+{
+    public function writeIdColumnGetter(WriterInterface $writer, string $nativeType)
+    {
+        $primaryKeyCalledId = false;
+        /** @var Column $column */
+        $column = $this;
+        if($column->isPrimary()) {
+            //@TODO - assumes that we only have one primary key column. If multiple then we have an issue...
+            if(strtolower($column->getColumnName()) !== 'id') {
+                $column->writePrimaryKeyFetcher($writer, $nativeType, 'getId');
+            }
+        }
+    }
+
+    public function writePrimaryKeyFetcher(WriterInterface $writer, string $nativeType, string $getterName = 'fetchId') {
```

Comment:
param $getterName could be mandatory

Reply:
Will be reworked very soon anyway when we deprecate fetchId